test/gtest/ucp/test_ucp_rma.cc

```diff
@@ -7,6 +7,153 @@
+                                  m_remote_mem_buf_memh,
+                                  m_remote_mem_buf_rkey);
+    }
+
+    void cleanup() {
+        ucs_status_t status;
+        ucp_rkey_destroy(m_remote_mem_buf_rkey);
+        status = ucp_mem_unmap(receiver().ucph(), m_remote_mem_buf_memh);
+        ASSERT_UCS_OK(status);
+        disconnect(sender());
+        delete m_local_mem_buf;
+        delete m_remote_mem_buf;
+        test_ucp_memheap::cleanup();
+    }
+
+    static ucp_params_t get_ctx_params() {
+        ucp_params_t params = ucp_test::get_ctx_params();
+        params.features |= UCP_FEATURE_RMA;
+        return params;
+    }
+
+    std::vector<ucp_test_param>
+    static enum_test_params(const ucp_params_t& ctx_params,
+                            const std::string& name,
+                            const std::string& test_case_name,
+                            const std::string& tls)
+    {
+        std::vector<std::pair<ucs_memory_type_t,
+                              ucs_memory_type_t> > mem_type_pairs =
+            mem_buffer::supported_mem_type_pairs();
+        std::vector<ucp_test_param> result;
+        int count = 0;
+
+        for (std::vector<std::pair<ucs_memory_type_t,
+                                   ucs_memory_type_t> >::const_iterator iter =
+                 mem_type_pairs.begin(); iter != mem_type_pairs.end(); ++iter) {
+            generate_test_params_variant(ctx_params, name, test_case_name + "/" +
+                                         std::string(ucs_memory_type_names[iter->first]) +
+                                         "<->" + std::string(ucs_memory_type_names[iter->second]),
+                                         tls, count++, result);
+        }
+
+        return result;
+    }
+
+protected:
+    static ucs_log_func_rc_t
+    rma_error_handler(const char *file, unsigned line, const char *function,
+                      ucs_log_level_t level, const char *message, va_list ap)
+    {
+        // Ignore errors that invalid input parameters as it is expected
+        if (level == UCS_LOG_LEVEL_ERROR) {
+            std::string err_str = format_message(message, ap);
+
+            if ((err_str.find(rma_err_exp_str) != std::string::npos) ||
+                /* the error below occurs when RMA lane can be configured for
+                 * current TL (i.e. RMA emulation over AM lane is not used) and
+                 * UCT is unable to do registration for the current memory type
+                 * (e.g. SHM TLs or IB TLs w/o GPUDirect support)*/
+                (err_str.find("remote memory is unreachable") != std::string::npos)) {
+                UCS_TEST_MESSAGE << err_str;
+                return UCS_LOG_FUNC_RC_STOP;
+            }
+        }
+
+        return UCS_LOG_FUNC_RC_CONTINUE;
+    }
+
+    void check_rma_op_status(ucs_status_t status) const {
+        if ((m_local_mem_type != UCS_MEMORY_TYPE_HOST) ||
+            (m_remote_mem_type != UCS_MEMORY_TYPE_HOST)) {
```

Comment:
should not check remote mem type

Reply:
we should not check remote memory type in case of GPUDirect only
otherwise (non-GPUDirect TL), a target side could register CUDA buffer and give it to an RMA initiator - but RMA initiator has to fail such transfer as it unsupported by UCP RMA, right?

Reply:
if target does not support GPU-D it would not have registered the memory with IB